stable_baselines/gail/dataset/record_expert.py

```diff
@@ -101,19 +106,33 @@ def generate_expert_traj(model, save_path, env=None, n_timesteps=0,
             observations.append(obs)
 
         if isinstance(model, BaseRLModel):
-            action, _ = model.predict(obs)
+            action, _ = model.predict(obs, mask=mask)
         else:
             action = model(obs)
 
         obs, reward, done, _ = env.step(action)
 
+        # Use only first env
+        if is_vec_env:
+
+            mask = [reward[0] for _ in range(env.num_envs)]
+            action = np.array([action[0]])
+            reward = np.array([reward[0]])
+            done = np.array([done[0]])
+
         actions.append(action)
         rewards.append(reward)
         episode_starts.append(done)
         reward_sum += reward
         idx += 1
         if done:
-            obs = env.reset()
+
+            if is_vec_env:
```

Comment:
why do you make a distinction here ?

Reply:
The Problem was that, it wood run in to errors when resetting multiple vector envs. I don't know why this is but, just letting the vector env reset it self, seems to fix it

Reply:
>it wood run in to errors when resetting multiple vector envs.

Can you open an issue with a minimal code (+ traceback) for that? Might be a bug.